backend/clickhouse/logs_test.go

```diff
@@ -539,6 +553,20 @@ func TestReadLogsWithBodyFilter(t *testing.T) {
 	}, Pagination{})
 	assert.NoError(t, err)
 	assert.Len(t, payload.Edges, 1)
+
+	payload, err = client.ReadLogs(ctx, 1, modelInputs.LogsParamsInput{
+		DateRange: makeDateWithinRange(now),
+		Query:     "STRIPE_INTEGRATION_ERROR", // uses reserved separator character
+	}, Pagination{})
+	assert.NoError(t, err)
+	assert.Len(t, payload.Edges, 2)
```

Comment:
This matches both logs, (`STRIPE_INTEGRATION_ERROR` and `STRIPE-INTEGRATION-ERROR`). This is an artifact of how `hasToken` works. 

The trade off is basically: gain perf win but at the possibility of matching more than expected.

I think we should optimize for this performance gain (since that gets the most value of searches) and if people dislike the overmatching, we can advise them to take advantage of structured logging more.